store/multistoreproof.go

```diff
@@ -56,6 +61,10 @@ func VerifyMultiStoreCommitInfo(storeName string, storeInfos []storeInfo, appHas
 
 // VerifyRangeProof verify iavl RangeProof
 func VerifyRangeProof(key, value []byte, substoreCommitHash []byte, rangeProof *iavl.RangeProof) error {
+	// Both rangeProof and substoreCommitHash are nil
```

Comment:
Hmm when do we expect this? What if just one is nil?

Reply:
When the query store is empty, both `rangeProof ` and `substoreCommitHash` will be nil. 
If only `rangeProof` is nil, an [invalid proof](https://github.com/tendermint/iavl/blob/3acc91fb8811db2c5409a855ae1f8e441fe98e2d/proof_range.go#L173) error will be returned.
If only `substoreCommitHash`, the proof verification will report error about root hash mismatching.